content/shmem_atomic_fetch_add_nbi.tex

```diff
@@ -0,0 +1,58 @@
+\color{Green}
+\apisummary{
+    The nonblocking atomic routine performs an atomic fetch-and-add operation on
+    a remote data object.
+}
+
+\begin{apidefinition}
+
+\begin{C11synopsis}
+void @\FuncDecl{shmem\_atomic\_fetch\_add\_nbi}@(TYPE *fetch, TYPE *dest, TYPE value, int pe);
+void @\FuncDecl{shmem\_atomic\_fetch\_add\_nbi}@(shmem_ctx_t ctx, TYPE *fetch, TYPE *dest, TYPE value, int pe);
+\end{C11synopsis}
+where \TYPE{} is one of the standard \ac{AMO} types specified by
+Table~\ref{stdamotypes}.
+
+\begin{Csynopsis}
+void @\FuncDecl{shmem\_\FuncParam{TYPENAME}\_atomic\_fetch\_add\_nbi}@(TYPE *fetch, TYPE *dest, TYPE value, int pe);
+void @\FuncDecl{shmem\_ctx\_\FuncParam{TYPENAME}\_atomic\_fetch\_add\_nbi}@(shmem_ctx_t ctx, TYPE *fetch, TYPE *dest, TYPE value, int pe);
+\end{Csynopsis}
+where \TYPE{} is one of the standard \ac{AMO} types and has a corresponding
+\TYPENAME{} specified by Table~\ref{stdamotypes}.
+
+\begin{apiarguments}
+
+  \apiargument{IN}{ctx}{A context handle specifying the context on which to
+      perform the operation. When this argument is not provided, the operation
+      is performed on the default context.}
+  \apiargument{OUT}{fetch}{Local data object to be updated.}
+  \apiargument{OUT}{dest}{The remotely accessible data object to be updated on
+      the remote \ac{PE}.}
+  \apiargument{IN}{value}{The value to be atomically added to \VAR{dest}.}
+  \apiargument{IN}{pe}{An integer that indicates the \ac{PE} number on which
+      \VAR{dest} is to be updated.}
+
+\end{apiarguments}
+
+\apidescription{
+    The nonblocking \FUNC{shmem\_atomic\_fetch\_add\_nbi} routines perform an
+    atomic fetch-and-add operation. An atomic fetch-and-add operation fetches
+    the old \VAR{dest} and adds \VAR{value} to \VAR{dest} without the
+    possibility of another atomic operation on the \VAR{dest} between the time
+    of the fetch and the update. This routine returns after posting the
```

Comment:
@jamesaross Noted that the sentence starting with "An atomic ..." is different from the other NBI routines and could potentially be seen as redefining atomicity for this operation.  Ensure wording is similar to other NBI operations.

Reply:
fixed